include/RouteMap.h

```diff
@@ -305,57 +324,119 @@ struct RouteMapPosition {
           grib(nullptr), grib_is_data_deficient(false) {} /* avoid waiting forever in update longitudes */
     bool Update();
 
     wxString RouteGUID;       /* Route GUID if any */
-    wxString Start;
+    wxString Start;           /* The name of starting position, which is resolved to StartLat/StartLon. */
     wxString StartGUID;
 
-    wxString End;
+    wxString End;             /* The name of the destination position, which is resolved to EndLat/EndLon. */
     wxString EndGUID;
 
-    wxDateTime StartTime;
+    wxDateTime StartTime;     /* The time when the boat leaves the starting position. */
 
     double DeltaTime; /* default time in seconds between propagations */
     double UsedDeltaTime; /* time in seconds between propagations */
 
-    Boat boat;
-    wxString boatFileName;
+    Boat boat;                /* The polars of the boat, used for the route calculation. */
+    wxString boatFileName;    /* The name of the boat XML file referencing polars. */
     
     enum IntegratorType { NEWTON, RUNGE_KUTTA } Integrator;
 
-    double MaxDivertedCourse, MaxCourseAngle, MaxSearchAngle, MaxTrueWindKnots, MaxApparentWindKnots;
-    double MaxSwellMeters, MaxLatitude, TackingTime, WindVSCurrent;
+    // The maximum angle the boat can be diverted from the bearing to the destination,
+    // at each step of the route calculation, in degrees.
+    // This represents the angle away from Start to End bearing (StartEndBearing).
+    // The normal setting is 100 degrees, which speeds up calculations.
+    // If the route needs to go around land, islands or peninsulas, the user can increase the value.
+    // E.g. the boat may have to go in the opposite direction then back to the destination bearing.
+    double MaxDivertedCourse;
+    // The maximum deviation away from the direct route.
+    double MaxCourseAngle;
+    // The maximum angle at each step of the route calculation.
+    double MaxSearchAngle;
+    // The calculated route will avoid a path where the true wind is above this value in knots.
+    double MaxTrueWindKnots;
+    // The calculated route will avoid a path where the gust wind is above this value in knots.
+    double MaxApparentWindKnots;
+
+    // The calculated route will avoid swells larger than this value in meters.
+    // If the grib data does not contain swell information, the maximum swell value is ignored.
+    // If there is no route within the maximum swell value, the route calculation will fail.
+    double MaxSwellMeters;
+    // The calculated route will not go beyond this latitude, as an absolute value.
+    // If the starting or destination position is beyond this latitude, the route calculation will fail.
+    double MaxLatitude;
+    // The penalty time to tack the boat, in seconds.
+    // The penalty time is added to the route calculation for each tack.
+    double TackingTime;
+    // Balance the influence of the wind and the ocean current on the route calculation.
+    double WindVSCurrent;
```

Comment:
I'm unclear what is the purpose of `WindVSCurrent`. I would like to improve the code comments of this field.

```c++
    if(configuration.WindVSCurrent) {
        /* Calculate the wind vector (Wx, Wy) and ocean current vector (Cx, Cy). */
        /* these are already computed in OverWater.. could optimize by reusing them */
        double Wx = VW*cos(deg2rad(W)), Wy = VW*sin(deg2rad(W));
        double Cx = VC*cos(deg2rad(C) + M_PI), Cy = VC*sin(deg2rad(C) + M_PI);

        if(Wx*Cx + Wy*Cy + configuration.WindVSCurrent < 0)
            return false;
    }
```

Reply:
It has to do with avoiding areas where the dot product of the wind vector and current vector have a strong negative value (sharp waves)

For example, 20 knot north wind, with 3 knot current from the south, the algorithm would avoid routing here if WindVSCurrent is 60 or less.